packages/wallet/src/rollup-aggregator.ts

```diff
@@ -210,111 +237,185 @@ export class RollupAggregator extends SimpleServer {
    */
   private async requestFaucetFunds(
     signedTransaction: SignedTransaction
-  ): Promise<SignedTransactionReceipt> {
-    if (!isFaucetTransaction(signedTransaction.transaction)) {
-      throw Error('Cannot handle non-Faucet Request in faucet endpoint')
-    }
-    const messageSigner: Address = this.signatureVerifier.verifyMessage(
-      serializeObject(signedTransaction.transaction),
-      signedTransaction.signature
-    )
-    if (messageSigner !== signedTransaction.transaction.requester) {
-      throw Error('Faucet requests must be signed by the request address')
-    }
+  ): Promise<SignedStateReceipt> {
+    try {
+      if (!isFaucetTransaction(signedTransaction.transaction)) {
+        throw Error('Cannot handle non-Faucet Request in faucet endpoint')
+      }
+      const messageSigner: Address = this.signatureVerifier.verifyMessage(
+        serializeObject(signedTransaction.transaction),
+        signedTransaction.signature
+      )
+      if (messageSigner !== signedTransaction.transaction.sender) {
+        throw Error(
+          `Faucet requests must be signed by the request address. Signer address: ${messageSigner}, sender: ${signedTransaction.transaction.sender}`
+        )
+      }
 
-    // TODO: Probably need to check amount before blindly giving them this amount
+      // TODO: Probably need to check amount before blindly giving them this amount
 
-    const { requester, amount } = signedTransaction.transaction
-    // Generate the faucet txs (one sending uni the other pigi)
-    const faucetTxs = await generateFaucetTxs(
-      requester,
-      amount,
-      this.wallet.address,
-      this.signatureProvider
-    )
+      const { sender, amount } = signedTransaction.transaction
+      // Generate the faucet txs (one sending uni the other pigi)
+      const faucetTxs = await generateFaucetTxs(
+        sender,
```

Comment:
Can’t remember why I did this, but I think there was a reason... I think consistency or something that made the `respond(...)` function easier. Will take a look and update if it’s easy.

Reply:
Ohh yeah, now that I see it, this is just  the original tx sender. It's  labeled recipient in the `generateFaucetTxs(...)` function as well as in the resulting transactions. This will get updated when we change how faucet requests take place, so leaving  for now.